fuel/datasets/binarized_mnist.py

```diff
@@ -1,19 +1,14 @@
 # -*- coding: utf-8 -*-
 import logging
 import os
-from collections import OrderedDict
-
-import numpy
 
 from fuel import config
-from fuel.datasets import IndexableDataset
-from fuel.utils import do_not_pickle_attributes
+from fuel.datasets import H5PYDataset
 
 logger = logging.getLogger(__name__)
 
 
-@do_not_pickle_attributes('indexables')
-class BinarizedMNIST(IndexableDataset):
+class BinarizedMNIST(H5PYDataset):
```

Comment:
I know that @dwf was quite keen on delegating to instead of inheriting from `H5PYDataset`. It's true that delegating might be beneficial if we later on want to give people the option to switch between `h5py` and PyTables, but I'm not sure how he envisioned that in detail though.